src/components/Fetcher/index.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Spinner from 'react-spinkit'
+import { connect } from 'react-redux'
+
+import LANGUAGE_FETCHER from 'endpoints/language'
+import PAGE_ENDPOINT from 'endpoints/page'
+import DISCLAIMER_FETCHER from 'endpoints/disclaimer'
+import LOCATION_FETCHER from 'endpoints/location'
+
+import Error from 'components/Error'
+
+import style from './style.css'
+
+function createFetcher (endpoint,
+                        createUrlOptions, createUrlTransformOptions,
+                        mapStateToProps = () => { return {} },
+                        shouldUpdate = () => false) {
+  let stateName = endpoint.name
+  let payloadName = stateName + 'Payload'
+
+  function mapStateToFetcherProps (state) {
+    let props = mapStateToProps(state)
+
+    props[payloadName] = state[stateName]
+    return props
+  }
+
+  let Fetcher = class extends React.PureComponent {
+    static propTypes = {
+      hideError: PropTypes.bool,
+      hideSpinner: PropTypes.bool
+    }
+
+    fetch (props) {
+      this.props.dispatch(endpoint.fetchEndpointAction(createUrlOptions(props), createUrlTransformOptions(props)))
+    }
+
+    invalidate () {
+      this.props.dispatch(endpoint.invalidateAction())
+    }
+
+    componentWillUnmount () {
+      this.invalidate()
+    }
+
+    componentWillMount () {
+      this.fetch(this.props)
+    }
+
+    componentWillUpdate (nextProps) {
+      if (shouldUpdate(this.props, nextProps)) {  // todo: this will need some more work to test -> an other issue as
+        // this is getting too big
+        this.fetch(nextProps)
+      }
+    }
+
+    render () {
+      let payload = this.props[payloadName]
+
+      if (!this.props.hideError && payload.error) {
+        return <Error error={payload.error}/>
+      }
+
+      if (payload.ready()) {
+        let newProps = {}
+        newProps[stateName] = payload.data
+        newProps[payloadName] = payload
+
+        return (
+          <div>
+            {
+              React.Children.map(this.props.children,
+                (child) => React.cloneElement(child, Object.assign({}, this.props, newProps)))
+            }
+          </div>
+        )
```

Comment:
Is it gone yet?
Also according to the Higher-order-component conventions (HOC) [(click)](https://facebook.github.io/react/docs/higher-order-components.html) our endpt.withFetcher() is not an HOC per def, because an HOC has a component as an argument that the return-component renders. Did you look into that doc?